docs/en/week07/07-3.md

```diff
@@ -0,0 +1,116 @@
+# Application of Autoencoders
+## DALL-E: Creating Images from Text
+
+A neural network that creates images from text captions. 
+![](https://i.imgur.com/FosonFM.jpg)
+
+
+Go to the [website](https://openai.com/blog/dall-e/) and play with the captions! 
+
+# Autoencoder
+Let's start with some definitions:
+
+## Definitions
+### Input
+$x$: is observed during both training and testing 
+
+$y$: is observed during training but not testing
+
+$z$: is not observed (neither during training nor during testing).
+
+### Output
+$h$: is computed from the input (hidden/internal)
+
+$\tilde{y}$: is computed from the hidden (predicted y, ~ means $circa$)
+
+Confused?
+Refer to the below figure to understand the use of different variables in different machine learning techniques.
+![](https://i.imgur.com/Y4CFlZc.png)
+
+## Introduction
+
+These kind of networks are used to learn the internal structure of an input and encode it in a hidden internal representation ($h$) which expresses the input.
+
+We already learned how to train energy-based models, let's look at the below network:
+![](https://i.imgur.com/NAUmcJd.png)
+
+Here instead of computing the minimization of the energy $E$ with respect to $z$, we use an encoder that approximates the minimization and provides a hidden representation $h$ for a given $y$.
+$$
+h = Encoder(y)
+$$
+
+Then the hidden representation is converted into $\tilde{y}$ (Here we don't have a predictor, we have an encoder)
+$$
+\tilde{y}= Decoder(h)
+$$
+
+Basically, $h$ is a squashing function $f$ of the rotation of our input $y$. ($y$ is the observation) and $\tilde{y}$ is a squashing function $g$ of the rotation of our hidden representation $h$.
+$$
+h = f(W_hy + b_h) \\
+\tilde{y} = g(W_yh + b_y)
+$$
+
+Note that, here $y$ and $\tilde{y}$ both belong to the same input space, and $h$ belong to $\mathbb{R}^d$ which is the internal representation. $W_h$ and $W_y$ are matrices for rotation.
+
+$$
+y, \tilde{y} \in \mathbb{R}^n \\
+h \in \mathbb{R}^d \\
+W_h \in \mathbb{R}^{d \times n} \\
+W_y \in \mathbb{R}^{n \times d}
+$$
+
+This is called Autoencoder. Enocder is performeing amortizing and we dont have to minimize the enerzy  $E$ but $F$:
+
+$$
+F(y) = C(y,\tilde{y}) + R(h)
+$$
+
+### Reconstruction Costs
+Below are the two examples of reconstruction energies:
+##### Real Valued Input:
+$$
+C(y,\tilde{y}) = ||y-\tilde{y}||^2 = ||y-Dec[Enc(y)]||^2
```

Comment:
The norm operator is `\Vert`.
**All vectors** should be encapsulated in `\vect{}`.

Reply:
Done